encode_utils/connection.py

```diff
@@ -1128,6 +1128,9 @@ def patch(self, payload, raise_403=True, extend_array_values=True):
                     # ['/documents/id1', 'id1']
                     # such a duplicate should be identified and removed, leaving us with ["id1"].
                     # Checks for arrays of strings or of dicts.
+                    if len(val) == 0:
+                        payload.pop(key)
+                        continue
```

Comment:
After testing this out, I think we need to move this section before line 1121. If the object already has some values for that key on the portal ('/documents/123'), val.extend() will add an empty string to the array, which then fails validation.

Reply:
Error I was encountering comes from create_payloads_from_tsv() in eu_register, so the current code should work as intended. No changes needed.